src/desktop_develop.ts

```diff
@@ -201,21 +224,33 @@ async function pruneBuilds(dir, exp) {
     }
 }
 
-class DesktopDevelopBuilder {
-    constructor(winVmName, winUsername, winPassword, rsyncRoot) {
-        this.winVmName = winVmName;
-        this.winUsername = winUsername;
-        this.winPassword = winPassword;
-        this.rsyncRoot = rsyncRoot;
-
-        this.pubDir = path.join(process.cwd(), 'packages.riot.im');
-        // This should be a reprepro dir with a config redirecting
-        // the output to pub/debian
-        this.debDir = path.join(process.cwd(), 'debian');
-        this.appPubDir = path.join(this.pubDir, 'nightly');
-    }
+function rm(path: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+        rimraf(path, (err) => {
+            err ? reject(err) : resolve();
+        });
+    });
+}
 
-    async start() {
+export default class DesktopDevelopBuilder {
+    private pubDir = path.join(process.cwd(), 'packages.riot.im');
+    // This should be a reprepro dir with a config redirecting
+    // the output to pub/debian
+    private debDir = path.join(process.cwd(), 'debian');
+    private appPubDir = path.join(this.pubDir, 'nightly');
+    private building = false;
+    private riotSigningKeyContainer: string;
+    private lastBuildTimes = {} as Record<Type, number>;
+    private lastFailTimes = {} as Record<Type, number>;
```

Comment:
preferred by our code style (informal, probably should be formal):
```suggestion
    private lastBuildTimes: Record<Type, number> = {};
    private lastFailTimes: Record<Type, number> = {};
```

Reply:
That's what I tried first, but you get the error `Type '{}' is missing the following properties from type 'Record<Type, number>': win32, win64, mac, linux`. It seems like without the `as` cast, it expects _all_ keys to be present in the value, which isn't true for just `{}`.

Reply:
You could maybe use `Partial<Record<Type, number>>`

Reply:
Ah yeah, that works, thanks!

Reply:
You resolved this but don't appear to have pushed anything?